### PR TITLE
Chapter 8 login

### PR DESCRIPTION
**Pull request by Arizar98**

### CODE REVIEW

**Comment by Arizar98** [comment]
Ready

File: app/assets/stylesheets/sessions.scss [hunk 1]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Sessions controller here.
+// They will automatically be included in application.css.
+// You can use Sass (SCSS) here: https://sass-lang.com/
```
Comment (saobv-2223) [inline app/assets/stylesheets/sessions.scss]: remove file that never used

File: app/controllers/sessions_controller.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
+class SessionsController < ApplicationController
+  def new
+  end
```
Comment (saobv-2223) [inline app/controllers/sessions_controller.rb]:
```suggestion
  def new; end
```

File: app/controllers/sessions_controller.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+  def create
+    user = User.find_by(email: params[:session][:email].downcase)
```
Comment (saobv-2223) [inline app/controllers/sessions_controller.rb]:
```suggestion
    user = User.find_by email: params[:session][:email].downcase
```

File: app/controllers/sessions_controller.rb [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+      render :new
+    end
+  end
+
+  def destroy
+    log_out
+    redirect_to root_path
+  end
+
+end
```
Comment (saobv-2223) [inline app/controllers/sessions_controller.rb]:
```suggestion
  end
end
```

File: app/controllers/users_controller.rb [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+      flash[:success] = t "welcome_to_the_sample_app!"
+      redirect_to @user
+    else
+      render :new
+      flash[:alert] = t ".try_again"
+    end
+  end
+
+  private
+  def user_params
```
Comment (saobv-2223) [inline app/controllers/users_controller.rb]:
```suggestion
  private
  
  def user_params
```

File: app/helpers/sessions_helper.rb [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+  end
+
+  # Returns true if the user is logged in, false otherwise.
+  def logged_in?
+    current_user.present?
+  end
+
+  # Logs out the current user.
+  def log_out
+    session.delete(:user_id)
```
Comment (saobv-2223) [inline app/helpers/sessions_helper.rb]:
```suggestion
    session.delete :user_id
```

File: config/credentials.yml.enc [hunk 7]
```diff
@@ -0,0 +1 @@
+rZnsNJXszaYhS95au6gMs9jLs6sMReyzm9u93Cfbcd2Os9TekUfdbgfe+uFMVWFdUmpJq84TdTXCNHpUY0KhK1E+vMAtIX5KL0MykVuqMy28LeC3B7KjP2n6JF7O/z4NUBW25MfEx/UTEQWR7oipRa/aIj34NmoD5CJ7IYYvxdoNl877VSqefhlIOofpR/XEu+RLoZT9YT/GUWjYYBLkEKk5GakplKqzTjCOTqkPJchI5K92Tluukb+HFjNLER05JiUiBpopZtVyZDpO/IYG6lOk6wFbY4yskg8zawd9xMnN0m00vd9SForaOiPwznYDTTZ6gu/yiLJzJOvM2UIohCHXlGvrlRzwClyJrJSvwAbJNSHw0A9pNTaTdi+2Rave+yERzaNev4+uHTgN4UYIIf1HqGtcCG0kbR+5--K6DoXjWzJUuAW7hM--GAco5RzmMZNdmxNh4unJ9w==
```
Comment (saobv-2223) [inline config/credentials.yml.enc]: image

**Comment by Arizar98** [comment]
Ready

**Review by saobv-2223 — COMMENTED** [review]
Your PR is having 3 commits, merge to 1 commit please

**Comment by Arizar98** [comment]
Ready

**Review by saobv-2223 — COMMENTED** [review]
Your pull request is having 3 commits, merge to 1 commit

**Comment by Arizar98** [comment]
Ready

**Comment by ngocvt-0484** [comment]
merged

**Pull request merged**